src/Raft.sol

```diff
@@ -78,47 +89,86 @@ contract Raft is
     _unpause();
   }
 
-  // we are basically implementing the functionality of ERC721URIStorage ourselves here
-  function setTokenURI(uint256 tokenId, string memory uri)
-    public
-    virtual
-    onlyOwner
-  {
+  /**
+   * @dev Sets the URI of the token with the given ID.
+   *      Reverts if the token does not exist.
+   * @param tokenId uint256 ID of the token to set its URI.
+   * @param uri string URI to assign to the token.
+   * @notice Only the owner of the contract can call this function.
+   */
+  function setTokenURI(
+    uint256 tokenId,
+    string memory uri
+  ) public virtual onlyOwner {
     require(_exists(tokenId), "setTokenURI: URI set of nonexistent token");
     _tokenURIs[tokenId] = uri;
 
     emit MetadataUpdate(tokenId);
   }
 
+  /**
+   * @dev Sets the admin status for a given tokenId and admin address.
+   * @param tokenId The ID of the token to set admin status for.
+   * @param admin The address of the admin whose status to set.
+   * @param isActive Whether the admin is active or not.
+   */
   function setAdmin(
     uint256 tokenId,
     address admin,
     bool isActive
   ) public virtual {
-    require(_exists(tokenId), "addAdmin: tokenId does not exist");
-    require(ownerOf(tokenId) == msg.sender, "addAdmin: unauthorized");
+    require(_exists(tokenId), "setAdmin: tokenId does not exist");
+    require(ownerOf(tokenId) == msg.sender, "setAdmin: unauthorized");
 
     _admins[tokenId][admin] = isActive;
 
     emit AdminUpdate(tokenId, admin, isActive);
   }
 
-  function isAdminActive(uint256 tokenId, address admin)
-    public
-    view
-    virtual
-    returns (bool)
-  {
+  /**
+   * @dev Sets multiple admins for a given tokenId.
+   * @param tokenId The token ID for which the admins are being set.
+   * @param admins An array of addresses representing the admins.
+   * @param isActive An array of booleans representing the admin status (active or not).
+   */
+
+  function setMultipleAdmins(
+    uint256 tokenId,
+    address[] memory admins,
+    bool[] memory isActive
+  ) public virtual {
+    require(_exists(tokenId), "setMultipleAdmins: tokenId does not exist");
+    require(ownerOf(tokenId) == msg.sender, "setMultipleAdmins: unauthorized");
+    require(
+      admins.length == isActive.length,
+      "setMultipleAdmins: admins and isActive arrays must have the same length"
+    );
+
+    for (uint256 i = 0; i < admins.length; i++) {
+      _admins[tokenId][admins[i]] = isActive[i];
```

Comment:
we safe all addresses in `_admins[tokenId]`, including those of non-admin users. does this come with any sort of penalty in terms of memory usage or cost?

Reply:
thx for clarifying! If there's no significant impact of keeping obsolete admin addresses in memory, we can also keep it.
quick question concerning [isAdminActive](https://github.com/otterspace-xyz/otterspace-contracts/pull/164/files#diff-33ca5ec33207d5d6cb91ae9a05227855baaa7a4cfa8fd7bcba853074053872f3R162): would this also return a boolean if `admin` has never been set before?

Reply:
yep